packages/common/src/filters/selectFilter.ts

```diff
@@ -398,7 +404,7 @@ export class SelectFilter implements Filter {
 
     // if there's a search term, we will add the "filled" class for styling purposes
     if (this.isFilled) {
```

Comment:
replace line 406 to 408 by 1 line
```diff
- if (this.isFilled) {
-    this.$filterElm?.addClass('filled').siblings('div .search-filter').addClass('filled');
- }
```
with this 1 line
```ts
this.updateFilterStyle(this.isFilled);
```
further below for the third and last part, replace lines 457 to 475
```diff
-  protected onTriggerEvent() {
-    if (this.$filterElm) {
-      const selectedItems = this.getValues();
-
-      if (Array.isArray(selectedItems) && selectedItems.length > 1 || (selectedItems.length === 1 && selectedItems[0] !== '')) {
-        this.isFilled = true;
-        this.$filterElm?.addClass('filled').siblings('div .search-filter').addClass('filled');
-      } else {
-        this.isFilled = false;
-        this.$filterElm.removeClass('filled');
-        this.$filterElm.siblings('div .search-filter').removeClass('filled');
-      }
-
-      this.searchTerms = selectedItems;
-      this.callback(undefined, { columnDef: this.columnDef, operator: this.operator, searchTerms: selectedItems, shouldTriggerQuery: this._shouldTriggerQuery });
-      // reset flag for next use
-      this._shouldTriggerQuery = true;
-    }
-  }
```
with this updated method and another new one
```ts
protected onTriggerEvent() {
    if (this.$filterElm) {
      const selectedItems = this.getValues();
      this.updateFilterStyle(Array.isArray(selectedItems) && selectedItems.length > 1 || (selectedItems.length === 1 && selectedItems[0] !== ''));
      this.searchTerms = selectedItems;
      this.callback(undefined, { columnDef: this.columnDef, operator: this.operator, searchTerms: selectedItems, shouldTriggerQuery: this._shouldTriggerQuery });
      // reset flag for next use
      this._shouldTriggerQuery = true;
    }
}

/** Set value(s) on the DOM element */
protected updateFilterStyle(isFilled: boolean) {
    if (isFilled) {
      this.isFilled = true;
      this.$filterElm?.addClass('filled').siblings('div .search-filter').addClass('filled');
    } else {
      this.isFilled = false;
      this.$filterElm.removeClass('filled');
      this.$filterElm.siblings('div .search-filter').removeClass('filled');
    }
}
```